sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1229,3 +1229,140 @@ case class Flatten(child: Expression) extends UnaryExpression {
+       ['123', '123']
+  """)
+case class ArrayRepeat(left: Expression, right: Expression)
+  extends BinaryExpression with ExpectsInputTypes {
+
+  override def dataType: ArrayType = ArrayType(left.dataType, left.nullable)
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType, IntegerType)
+
+  override def nullable: Boolean = right.nullable
+
+  override def eval(input: InternalRow): Any = {
+    val count = right.eval(input)
+    if (count == null) {
+      null
+    } else {
+      new GenericArrayData(List.fill(count.asInstanceOf[Int])(left.eval(input)))
+    }
+  }
+
+  override def prettyName: String = "array_repeat"
+
+  override def nullSafeCodeGen(ctx: CodegenContext,
+                               ev: ExprCode,
+                               f: (String, String) => String): ExprCode = {
+    val leftGen = left.genCode(ctx)
+    val rightGen = right.genCode(ctx)
+    val resultCode = f(leftGen.value, rightGen.value)
+
+    if (nullable) {
+      val nullSafeEval =
+        leftGen.code +
+          rightGen.code + ctx.nullSafeExec(right.nullable, rightGen.isNull) {
+            s"""
+              ${ev.isNull} = false;
+              $resultCode
+            """
+          }
+
+      ev.copy(code =
+        s"""
+           | boolean ${ev.isNull} = true;
+           | ${CodeGenerator.javaType(dataType)} ${ev.value} =
+           |   ${CodeGenerator.defaultValue(dataType)};
+           | $nullSafeEval
+         """.stripMargin
+      )
+    } else {
+      ev.copy(code =
+        s"""
+           | boolean ${ev.isNull} = false;
+           | ${leftGen.code}
+           | ${rightGen.code}
+           | ${CodeGenerator.javaType(dataType)} ${ev.value} =
+           |   ${CodeGenerator.defaultValue(dataType)};
+           | $resultCode
+         """.stripMargin
+        , isNull = FalseLiteral)
+    }
+
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+    nullSafeCodeGen(ctx, ev, (l, r) => {
+      val et = dataType.elementType
+      val isPrimitive = CodeGenerator.isPrimitiveType(et)
+
+      val arrayDataName = ctx.freshName("arrayData")
+      val arrayName = ctx.freshName("arrayObject")
+      val numElements = ctx.freshName("numElements")
+
+      val genNumElements =
+        s"""
+           | int $numElements = 0;
+           | if ($r > 0) {
+           |   $numElements = $r;
+           | }
+         """.stripMargin
+
+      val initialization = if (isPrimitive) {
+        val arrayName = ctx.freshName("array")
+        val baseOffset = Platform.BYTE_ARRAY_OFFSET
+        s"""
+           | int numBytes = ${et.defaultSize} * $numElements;
+           | int unsafeArraySizeInBytes =
+           |   UnsafeArrayData.calculateHeaderPortionInBytes($numElements)
+           |     + org.apache.spark.unsafe.array.ByteArrayMethods
+           |       .roundNumberOfBytesToNearestWord(numBytes);
+           | byte[] $arrayName = new byte[unsafeArraySizeInBytes];
```

Comment:
Do we throw an exception for the large number of elements with wider elements? For example, `0x7000_0000` long elements. I think that `GenericArrayData` can hold these elements.

Reply:
So you mean, we should do a size check to make sure it fits in the array, and if it doesn't we should do boxing and initialize a `GenericArrayData` instead with the given size?
Also when you say `0x7000_0000`, do you mean the hex value?